packages/flutter_tools/lib/src/windows/windows_device.dart

```diff
@@ -34,14 +35,25 @@ class WindowsDevice extends DesktopDevice {
       operatingSystemUtils: operatingSystemUtils,
   );
 
+  final OperatingSystemUtils _operatingSystemUtils;
+
   @override
   bool isSupported() => true;
 
   @override
   String get name => 'Windows';
 
   @override
-  Future<TargetPlatform> get targetPlatform async => TargetPlatform.windows_x64;
+  late final Future<TargetPlatform> targetPlatform = () async {
+    return synchronousTargetPlatform();
+  }();
+
+  TargetPlatform synchronousTargetPlatform() {
+    if (_operatingSystemUtils.hostPlatform == HostPlatform.windows_arm64) {
+      return TargetPlatform.windows_arm64;
+    }
+    return TargetPlatform.windows_x64;
+  }
```

Comment:
Please make the synchronous helper private with a `_` prefix.

Also what do you think of this version? I don't feel strongly, feel free to keep the current approach if you prefer it.

```suggestion
  Future<TargetPlatform> get targetPlatform async => _targetPlatform;

  TargetPlatform get _targetPlatform => switch (_operatingSystemUtils.hostPlatform) {
    HostPlatform.windows_arm64 => TargetPlatform.windows_arm64,
    _ => TargetPlatform.windows_x64,
  };

```

Reply:
Merged your version.